library/FILEBROWSER

```diff
@@ -159,6 +159,10 @@ You specify how many versions to keep.")))
                                                               )
                                                        ("Shape to Fit" FB.SHAPECOMMAND 
                                      "Widen or narrow the browser so that all information is visible"
+                                                              )
+                                                       ("Fix directory dates" FB.FIX-DIRECTORY-DATES
+                                                              
+                                                "Ensure that directory dates match FILECREATED dates"
```

Comment:
If "fix" is misleading, we could call it something else.  But all bets are off anyway if somebody edits with a text editor--they could even edit the filecreated date.  The sacred trust can be broken in many different ways

The point of this is to make it easy to fix things up when it is git hub or other inherited filedates that are out of step.

Another useful submenu item, perhaps under "delete", would be to detect byte-wise duplicates and mark them systematically for deletion.

Reply:
most of the duplicates come about whenever you switch branches ... it replaces files without updating the hard links.
So if you have a file FOO which in lisp has FOO;1 FOO;2 FOO;3 then FOO;3 is hard linked to FOO (no version). But when you switch branches in GitHub (and even if you switch back) it doesn't recreate the hard link. Rather from the Lisp side you'll see FOO;4 for what in Unix is FOO (no version) as well as FOO;1 FOO;2 FOO;3.

There's no easy way to fix this from Lisp since Lisp doesn't know about linked files.
The "restore-versions.sh" script restores the hard links.

One way to approach this is by careful use of ShellCommand applied to something like (UNIX-FILENAME-FOR-DSK ...)

Reply:
I made a new issue for fixing hard links after git actions.

Question: does FB now require COMPAREDIRECTORIES now to get FIX-DIRECTORY-DATES?

Reply:
No, it doesn't require comparedirectories.  The FIX-DIRECTORY-DATES there worried about more stuff.  This is very simple:  enumerate the browser pattern, examine and fix the creation dates as needed.

On duplicates, maybe there will be fewer going forward, and maybe the script will catch them all.  But many that I see are on older versions that are not linked to anything.  So it isn't the linking of the highest version, it's the residue that's left behind as new versions are created because of nonlinking somewhere in the past.